### PR TITLE
[RTE] Change rich text editor for composite to use boolean

### PR DESCRIPTION
**Pull request by palatter**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/src/composites/ChatComposite/ChatComposite.tsx [hunk 1]
```diff
@@ -87,12 +85,14 @@ export type ChatCompositeOptions = {
    */
   attachmentOptions?: AttachmentOptions;
 
-  /* @conditional-compile-remove(rich-text-editor-composite-support) @conditional-compile-remove(rich-text-editor) */
+  /* @conditional-compile-remove(rich-text-editor-composite-support) */
   /**
-   * Options to enable rich text editor for the edit box.
+   * enables rich text editor for the edit box
```
Comment (vhuseinova-msft) [inline packages/react-composites/src/composites/ChatComposite/ChatComposite.tsx:90]: Should we add any information about the send box too?

File: packages/react-composites/src/composites/CallWithChatComposite/CallWithChatComposite.tsx [hunk 2]
```diff
@@ -269,12 +267,14 @@ export type CallWithChatCompositeOptions = {
     hideSpotlightButtons?: boolean;
   };
 
-  /* @conditional-compile-remove(rich-text-editor) @conditional-compile-remove(rich-text-editor-composite-support) */
+  /* @conditional-compile-remove(rich-text-editor-composite-support) */
   /**
-   * Options to enable rich text editor for the edit box.
+   * enables rich text editor for the edit box
```
Comment (vhuseinova-msft) [inline packages/react-composites/src/composites/CallWithChatComposite/CallWithChatComposite.tsx:272]: Should we add any information about the send box too?

File: packages/react-composites/src/composites/ChatComposite/ChatComposite.tsx (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-composites/src/composites/ChatComposite/ChatComposite.tsx:90]: Also, should we start with an capital letter?

**Review by vhuseinova-msft — APPROVED** [review]
Added a few comments but otherwise looks good

**Review by Leah-Xia-Microsoft — APPROVED** [review]

**Pull request merged**